unity/src/main/java/io/delta/unity/UCCatalogManagedCommitter.java

```diff
@@ -0,0 +1,95 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.unity;
+
+import static io.delta.kernel.internal.util.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import io.delta.kernel.commit.CommitFailedException;
+import io.delta.kernel.commit.CommitMetadata;
+import io.delta.kernel.commit.CommitResponse;
+import io.delta.kernel.commit.Committer;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.utils.CloseableIterator;
+import io.delta.storage.commit.uccommitcoordinator.UCClient;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UCCatalogManagedCommitter implements Committer {
+  private static final Logger logger = LoggerFactory.getLogger(UCCatalogManagedCommitter.class);
+
+  private final UCClient ucClient;
+  private final String ucTableId;
+  private final Path tablePath;
+
+  public UCCatalogManagedCommitter(UCClient ucClient, String ucTableId, String tablePath) {
+    this.ucClient = requireNonNull(ucClient, "ucClient is null");
+    this.ucTableId = requireNonNull(ucTableId, "ucTableId is null");
+    this.tablePath = new Path(requireNonNull(tablePath, "tablePath is null"));
+  }
+
+  @Override
+  public CommitResponse commit(
+      Engine engine, CloseableIterator<Row> finalizedActions, CommitMetadata commitMetadata)
+      throws CommitFailedException {
+    requireNonNull(engine, "engine is null");
+    requireNonNull(finalizedActions, "finalizedActions is null");
+    requireNonNull(commitMetadata, "commitMetadata is null");
+    validateCommitMetadataLogPath(commitMetadata);
+
+    final CommitMetadata.CommitType commitType = commitMetadata.getCommitType();
+
+    if (commitType != CommitMetadata.CommitType.CATALOG_WRITE) {
+      // TODO: Support CATALOG_CREATE in the future, too
+      throw new UnsupportedOperationException("Unsupported commit type: " + commitType);
+    }
+
+    // TODO: lastKnownBackfilledVersion? Take that in as a hint.
+
+    if (commitMetadata.getNewProtocolOpt().isPresent()) {
+      // TODO: support this
+      throw new UnsupportedOperationException("Protocol change is not yet implemented");
+    }
+    if (commitMetadata.getNewMetadataOpt().isPresent()) {
+      // TODO: support this
+      throw new UnsupportedOperationException("Metadata change is not yet implemented");
+    }
+
+    // TODO: support this
+    throw new UnsupportedOperationException("Commit logic is not yet implemented");
```

Comment:
Are these each tied to a specific piece of implementation logic? (as the reason for this structure)

Reply:
I'd say: yes, each of these `Unsupported...` are coupled to some distrinct functionality.

I'm just trying to roll out and implement this class iteratively, and we can support one piece of logic without the others. e.g. we can implement the normal commit logic, without supporting the protocol/metadata changes (which requires mapping kernel protocol to the delta storage abstract protocol, for example)